signal/signal.go

```diff
@@ -147,3 +153,12 @@ func (c *Interceptor) RequestShutdown() {
 func (c *Interceptor) ShutdownChannel() <-chan struct{} {
 	return c.shutdownChannel
 }
+
+// GetExitCode fetches the exit code that has been set of the Interceptor
+func (c *Interceptor) GetExitCode() int {
```

Comment:
We _could_ have a data race here is `GetExitCode` is called at the same time as `setExitCode`. Shouldn't be an issue in practice because it only makes sense to call this after we've shutdown. 

Wondering whether we could create `exitCode = make(chan int, 1)` and then ensure that the interceptor always sends into it on exit? That way a `GetExitCode` can just read from the channel. Might be too much complexity though.